src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java

```diff
@@ -48,15 +45,15 @@ public void handle(Context ctx) throws Exception {
     ServerRequest request = new ServerRequestImpl(ctx.getRequest());
     final Span span = handler.handleReceive(extractor, request);
 
-    //place the Span in scope so that downstream code (e.g. Ratpack handlers
-    //further on in the chain) can see the Span.
-    final Tracer.SpanInScope scope = tracing.tracer().withSpanInScope(span);
     ctx.getResponse().beforeSend(response -> {
       ServerResponse serverResponse = new ServerResponseImpl(response, request, ctx.getPathBinding());
       handler.handleSend(serverResponse, null, span);
-      scope.close();
     });
-    ctx.next();
+    //place the Span in scope so that downstream code (e.g. Ratpack handlers
+    //further on in the chain) can see the Span.
+    try (Tracer.SpanInScope scope = tracing.tracer().withSpanInScope(span)) {
```

Comment:
good catch!

Reply:
something yelled in the logs :) my goal with the base test class is to make it scream as much as possible, as frankly instrumentation is very hard job of remembering and eagle eyes otherwise! cc @jcchavezs

Reply:
I would never have spotted that - out of curiosity, what do the logs say in this case (and would it warrant a test failure)?